dspace/cli/senior_thesis_community.rb

```diff
@@ -0,0 +1,354 @@
+module DSpace
+  module CLI
+    class SeniorThesisCollection
+      attr_reader :obj
+
+      def self.kernel
+        ::DSpace
+      end
+
+      def initialize(obj)
+        @obj = obj
+      end
+
+      def removeItem(item)
+        @obj.removeItem(item.obj)
+      end
+
+      def update
+        @obj.update
+        self.class.kernel.commit
+      end
+    end
+
+    class SeniorThesisWorkflowItem
+      attr_reader :obj
+
+      def self.kernel
+        ::DSpace
+      end
+
+      # This is some internal bug; I am not certain why I cannot use this from DItem
+      def initialize(obj)
+        @obj = obj
+      end
+
+      def delete
+        @obj.deleteWrapper
+        self.class.kernel.commit
+        @obj = nil
+      end
+    end
+
+    class SeniorThesisItem < ::DItem
+      java_import org.dspace.workflow.WorkflowItem
+      java_import org.dspace.content.Metadatum
+
+      attr_reader :obj
+
+      def self.kernel
+        ::DSpace
+      end
+
+      # This is some internal bug; I am not certain why I cannot use this from DItem
+      def initialize(obj)
+        @obj = obj
+      end
+
+      # This is some internal bug; I am not certain why I cannot use this from DItem
+      def getMetadataByMetadataString(metadata_field)
+        @obj.getMetadataByMetadataString(metadata_field)
+      end
+
+      def self.build_metadata(schema, element, qualifier = nil, language = nil)
+        metadata = Java::OrgDspaceContent::Metadatum.new
```

Comment:
I suspect that I will need to implement a new Class for `Metadatum` here which persists the metadata database rows.